File: 프로그래머스/lv1/12977. 소수 만들기/소수 만들기.js
function solution(nums) {

  let sum = []
  for (let i=0; i<nums.length-2; i++) {
    for (let j=i+1; j<nums.length-1; j++) {
      for (let k=j+1; k<nums.length; k++) {
        sum.push(nums[i]+nums[j]+nums[k])
      }
    }
  }

  // 각 요소 순회하면서, 각 요소가 소수인지를 체크
  let cnt = 0
  for (let i=0; i<sum.length; i++) {
    let isPrime = true

    for (let j=3; j<sum[i]; j++) {
      if (sum[i] % j === 0) {
        isPrime = false
        break
      }
    }
    if (isPrime===true) cnt++
  }

  return cnt
}

File: 프로그래머스/1/12977. 소수 만들기/소수 만들기.py
# 1. nums숫자중, 3개를 고르는 '조합' -> sum
# 2. 소수인지 판별
import math
from itertools import combinations

def is_prime(num):
    for i in range(2,int(math.sqrt(num))+1):
        if num % i == 0:
            return False
    return True

def solution(nums):
    cnt = 0
    perm = list(combinations(nums,3))

    for ele in perm:
        if is_prime(sum(ele)): # True면
            cnt += 1

    return cnt